test-apps/imodel-transformer/src/Main.ts

```diff
@@ -56,6 +56,17 @@ Transform the specified source iModel into a new target iModel.
 You must set up a .env file to connect to an online iModel, see the .env.template file to do so.
 `.trim();
 
+/** allows lazy access token access, so we don't need to get it if we don't need it */
+class LazyAccessToken {
```

Comment:
[@wgoehrig note all the confusion that making `getAccessToken` async has wrought.]

Just call `await IModelHubUtls.getAccessToken` everywhere and don't cache the result. If the session is logged in it will return a valid token, otherwise it returns "". AccessToken should never be undefined.

If the token needs to be refreshed (because your program runs for a while) it may change over time.

Reply:
makes sense, will make that change

Reply:
done

Reply:
Yes, we've (I've) done a poor job of documenting all of the authentication changes we made in 3.0. Let me try it out now and see if I can make clear the changes.

Reply:
We've had a mess with this whole concept (which has changed multiple times). As i now look at what's in IModelHubUtils, i see that *it* is wrong. It attempts to log in and that's just not right. You should call signIn somewhere at startup, wait for it, and then you should call `IModelHost.getAccessToken` everywhere else.

 @wgoehrig, @calebmshafer I plead one more time, can we PLEASE make IModelHost.getAccessToken sync. If *we* can't get this right, how is anyone else supposed to?

Reply:
I'm no longer caching the result of getAccessToken, but, pending comments, I'm currently going with a lazy async fetching of the access token anyway. `getAccessToken` is now called everywhere, sign in is called once, and therefore the token can be refreshed.

Reply:
Please don't call your method "getAccessToken". Maybe "acquireAccessToken"?

Reply:
this rename has been made